server/vulnerabilities/nvd/sync/utils.go

```diff
@@ -0,0 +1,88 @@
+package nvdsync
+
+import (
+	"archive/zip"
+	"bufio"
+	"compress/gzip"
+	"io"
+	"os"
+)
+
+func CompressFile(fileName string, newFileName string) error {
+	// Read old file
+	file, err := os.Open(fileName)
+	if err != nil {
+		return err
+	}
+	read := bufio.NewReader(file)
+	data, err := io.ReadAll(read)
+	if err != nil {
+		return err
+	}
+	file.Close()
+
+	// Write new file
+	newFile, err := os.Create(newFileName)
+	if err != nil {
+		return err
+	}
+	writer := gzip.NewWriter(newFile)
+	if _, err = writer.Write(data); err != nil {
+		return err
```

Comment:
It's usually safe to call `Close` twice , so we could add a `defer writer.Close()` and `defer newFile.Close()` before this (so that they are closed if this returns here).

Reply:
Sorry, I meant to add the `defer`s and leave the Close() invocations as is.